enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,181 @@
+Use of this existing framework also allows this new upgrade blocking mechanism to be bypassed by using the `force` upgrade option.
+
+The new method's logic will be driven by predefined key names each of which represents a potential minor level upgrade blocker.
+The key name will include an OCP major/minor verion number that indicates to which OCP version the key applies.
+Key names must be in the format `ack-X.Y-<description>` where `X.Y` is the major/minor version number and `<description>` is words describing to what the administrator is agreeing.
+For example, a key name beginning with `ack-4.8` would only apply to an OCP cluster currently at version 4.8.
+If running within the applicable version, CVO will check a newly defined configmap for the key.
+The key represents a specific upgrade blocking condition that must be acknowledged before a minor level upgrade is allowed.
+An administrator acknowledges a condition by adding the corresponding key with a value of `true` to the configmap.
+Values other than `true` will be ignored and therefore not clear the upgrade block.
+The keys will be named to make it explicitly clear what the administrator is acknowledging and thereby agreeing to.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to the next minor release until it finds each of the applicable predefined keys in the configmap with a value of `true`.
+For each key blocking an upgrade, CVO will generate a detailed message containing the key and pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion, i.e. entering the corresponding key into the configmap with a value of `true`.
+
+The new configmap `upgrade-ack` will be installed as part of the release in the namespace `openshift-cluster-version`.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+  name: upgrade-ack
+  namespace: openshift-cluster-version
+```
+It will initially have no keys since the administrator is expected to add the relevant key as part of acknowledgement.
+
+### 4.8.z Specific Implementation
+
+Clusters upgraded to 4.9 will no longer have access to these [APIs][k8s-1.22-removed-apis].
+This new upgrade blocking mechanism will be used in an attempt to require administrators to read detailed documentation regarding the removed APIs and the possible impact.
+This OCP documentation will list which APIs have been removed and how an administrator checks their cluster for use of those APIs.
+
+4.8.z versions that are upgradeable to 4.9 will contain this enhancement and have the configmap `openshift-cluster-version/upgrade-ack`.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to 4.9 until the administrator has added the key/value pair `ack-4.8-apis-being-removed-in-4.9: "true"` to the configmap.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    ack-4.8-apis-being-removed-in-4.9: "true"
```

Comment:
Any reason we're using an annotation here instead of a data key in the config map itself?

Reply:
i missed this, i would also have expected it to be data in the configmap.

Reply:
Agreed, should just be a key.

Reply:
Changed to use `data:`